src/utils/ExtractTestResults.ts

```diff
@@ -22,7 +25,7 @@ import { ValidationUtil } from './ValidationUtil';
  * @param record
  */
 export const extractMCTestResults = (record: DynamoDBRecord): MCRequest[] => {
-  const testResultUnmarshall = unmarshall(record.dynamodb.NewImage as any);
+  const testResultUnmarshall : TestResultSchema = unmarshall(record.dynamodb.NewImage as any) as TestResultSchema;
```

Comment:
is the any casting here necessary?

Reply:
unfortunately, yes. this is why the `// ts-ignore` was there previously. Throws the error:
```
TS2345: Argument of type { [key: string]: AttributeValue; } is not assignable to parameter of type
AttributeValue | Record<string, AttributeValue>
Property $unknown is missing in type { [key: string]: AttributeValue; } but required in type $UnknownMember
```

in pull test results, we got around this by doing `const currentRecord = unmarshall(dbRecord.dynamodb.NewImage as Record<string, AttributeValue>) as TestResultModel;`

unfortunately that doesn't work here for some reason, throws the same error as above.

Reply:
Why could it be that type, if anything; I would expect it to be TestResult | null. Makes me wonder if the type in the types definition is correct or not

Reply:
the unmarshaller returns a type of `AttributeValue | Record<string, AttributeValue>`